File: LeetCode_1957/Main.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    string makeFancyString(string s) {
        string res;
        for (int i = 0; i < s.size(); i++) {
            if (res.size() < 2) {
                res.push_back(s[i]);
            } else {
                int sz = res.size();
                if (res[sz - 1] == res[sz - 2] && res[sz - 1] == s[i]) {
                    continue;
                } else {
                    res.push_back(s[i]);
                }
            }
        }
        return res;
    }
};

File: LeetCode_1957/Main.java
class Solution {
    public String makeFancyString(String s) {
        StringBuilder res = new StringBuilder();
        for (int i = 0; i < s.length(); i++) {
            if (res.length() < 2) {
                res.append(s.charAt(i));
            } else {
                int sz = res.length();
                if (res.charAt(sz - 1) == res.charAt(sz - 2) && res.charAt(sz - 1) == s.charAt(i)) {
                    continue;
                } else {
                    res.append(s.charAt(i));
                }
            }
        }
        return res.toString();
    }
}